componenets/org.wso2.carbon.identity.application.authenticator.naver/src/main/java/org.wso2.carbon.identity.application.authenticator.naver/NaverCustomAuthenticator.java

```diff
@@ -159,4 +257,187 @@ public List<Property> getConfigurationProperties() {
     }
+
+    /**
+     * This method is responsible for obtaining authenticator-specific data needed to
+     * initialize the authentication process within the provided authentication context.
+     *
+     * @param context The authentication context containing information about the current authentication attempt.
+     * @return An {@code Optional} containing an {@code AuthenticatorData} object representing the initiation data.
+     *         If the initiation data is available, it is encapsulated within the {@code Optional}; otherwise,
+     *         an empty {@code Optional} is returned.
+     */
+    @Override
+    public Optional<AuthenticatorData> getAuthInitiationData(AuthenticationContext context) {
+
+        AuthenticatorData authenticatorData = new AuthenticatorData();
+        authenticatorData.setName(getName());
+        authenticatorData.setDisplayName(getFriendlyName());
+        authenticatorData.setI18nKey(getI18nKey());
+        String idpName = context.getExternalIdP().getIdPName();
+        authenticatorData.setIdp(idpName);
+
+        List<String> requiredParameterList = new ArrayList<>();
+        if (isTrustedTokenIssuer(context)) {
+            requiredParameterList.add(NaverCustomAuthenticatorConstants.ACCESS_TOKEN_PARAM);
+            requiredParameterList.add(ID_TOKEN_PARAM);
+            authenticatorData.setPromptType(FrameworkConstants.AuthenticatorPromptType.INTERNAL_PROMPT);
+            authenticatorData.setAdditionalData(getAdditionalData(context, true));
+        } else {
+            requiredParameterList.add(NaverCustomAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE);
+            requiredParameterList.add(NaverCustomAuthenticatorConstants.OAUTH2_PARAM_STATE);
+            authenticatorData.setPromptType(FrameworkConstants.AuthenticatorPromptType.REDIRECTION_PROMPT);
+            authenticatorData.setAdditionalData(getAdditionalData(context, false));
+        }
+        authenticatorData.setRequiredParams(requiredParameterList);
+
+        return Optional.of(authenticatorData);
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response, AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("processAuthenticationResponse");
+        }
+
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = (String)authenticatorProperties.get(CLIENT_ID);
+            String clientSecret = (String)authenticatorProperties.get(CLIENT_SECRET);
+            Boolean basicAuthEnabled = Boolean.parseBoolean((String)authenticatorProperties.get(IS_BASIC_AUTH_ENABLED));
+            String tokenEP = this.getTokenEndpoint(authenticatorProperties);
+            String token;
+            if (isTrustedTokenIssuer(context) && isNativeSDKBasedFederationCall(request)) {
```

Comment:
Hence isTrustedTokenIssuer method required only for Native mode, shall we change the AND operation execution order to avoid unnecessary tokenIssuer check on a non native mode scenario?